pkg/templates/machinecontroller/cloudprovider_specs.go

```diff
@@ -186,3 +186,31 @@ type AzureImageReference struct {
 	Sku       string `json:"sku,omitempty"`
 	Version   string `json:"version,omitempty"`
 }
+
+// VMWareCloudDirectorSpec represents VMware Cloud Director specific configuration.
+type VMWareCloudDirectorSpec struct {
```

Comment:
Why are some fields missing here? I see some additional fields in [the original struct](https://github.com/kubermatic/machine-controller/blob/master/pkg/cloudprovider/provider/vmware-cloud-director/types/types.go), such `PlacementPolicy` and similar.

Reply:
We don't have a way to test them at the moment since they are not enabled on our VCD setup. We can enable them after we test them.